Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -248,36 +284,39 @@ class FISTA(ISTA):
 
     """
 
-    def set_step_size(self, step_size):
-
-        """Set the default step size
+    def _calculate_default_step_size(self):
+        """Calculate the default step size if a step size rule or step size is not provided 
         """
 
-        if step_size is None:
-
-            if isinstance(self.f, ZeroFunction):
-                self._step_size = 1
+        if isinstance(self.f, ZeroFunction):
+            ret = 1
 
-            elif isinstance(self.f.L, Number):
-                self._step_size = 1./self.f.L
-
-            else:
-                raise ValueError("Function f is not differentiable")
+        elif isinstance(self.f.L, Number):
+            ret = 1./self.f.L
 
         else:
-            self._step_size = step_size
+            raise ValueError("Function f is not differentiable")
+        return ret
```

Comment:
Do you need this `else`? If the function cannot calculate the Lispschitz constant it'll throw an error.